### ISSUE TITLE
[BUG] HAT classifier not growing deeper than tree_depth=1, instead overriding previous split rule

### ISSUE THREAD

**Issue by fwille** [scikit-multiflow/scikit-multiflow#163]
**Describe the bug**
I noticed performance differences of the HAT classifier between MOA and skmultiflow implementations. My analysis of the problem so far has revealed, that the HAT classifier in skmultiflow never grows deeper than 1 and despite finding new split rules only replaces the previous split rule.

In MOA on the other hand, the HAT is growing using as example SEAGenerator Data Stream with instanceRandomSeed=1:
![generators SEAGenerator](https://user-images.githubusercontent.com/17760164/69739327-cee1cf00-1137-11ea-9543-6f8cc6511316.png)

**To Reproduce**

This MWE for skmultiflow shows, that although the number of nodes is increasing, the model is stuck on a tree depth of 1 but with changing split rules:
```
from skmultiflow.data import SEAGenerator
from skmultiflow.trees import HAT

stream = SEAGenerator(random_state=1)
stream.prepare_for_use()

hat= HAT()

for i in range(0,10000):
    X,y = stream.next_sample()
    hat.partial_fit(X, y)
    hat.predict(X)
    print(hat.get_model_measurements)
    print(hat.get_rules_description())
```

**Expected behavior**
Tree should be able to grow deeper than 1.

**Additional context**
This bug also leads to fluctuating performance of the HAT classifier at times where a split rule is replaced by another split rule with significantly different selectivity.

**Comment by smastelini** [comment]
Thank you @fwille for investigating this problem.

I started investigating the issue already. This new MWE ought to simplify the tracking for the possible bug.

@jacobmontiel, do you have any clues about the roots of this problem?

**Comment by smastelini** [comment]
Hello everyone. After a lot of comparisons between skmultiflow and MOA, I was able to find the root of the problem.

The culprit was related to a single line:

https://github.com/scikit-multiflow/scikit-multiflow/blob/f32719662bba6e98dd684af85e800c64d828bd6d/src/skmultiflow/trees/nodes/ada_split_node.py#L122-L125

At the line `125`, the `AdaSplitNode` has to pass its own reference and the `child_branch` to the `learn_from_instance` method. Currently, it just repeats the `parent` and `parent_branch` parameters.

My local tests after changing that line have shown that HAT seemingly starts behaving as expected.

I will create a PR to address this bug.

Thank you again @fwille for pointing out this problem.

**Comment by smastelini** [comment]
I created PR #164 to address this problem.

**Comment by jacobmontiel** [comment]
I am closing this issue, the corresponding fix has been merged in 029d85d547346a3c04c5287d4ddc42cf84865491
